File: AppListaCurso/app/src/main/java/devandroid/thalles/applistacurso/view/MainActivity.java
package devandroid.thalles.applistacurso.view;

import androidx.appcompat.app.AppCompatActivity;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;
import devandroid.thalles.applistacurso.R;
import devandroid.thalles.applistacurso.controller.PessoaController;
import devandroid.thalles.applistacurso.model.Pessoa;

public class MainActivity extends AppCompatActivity {

    Pessoa pessoa;
    PessoaController pessoaController;

    EditText editPrimeiroNome;
    EditText editSobrenome;
    EditText editCursoDesejado;
    EditText editNumeroContato;
    Button btnLimpar;
    Button btnSalvar;
    Button btnFinalizar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Variáveis
        pessoa = new Pessoa();
        pessoaController = new PessoaController( MainActivity.this );

        pessoaController.buscar(pessoa);

        // Mapeando os campos
        editPrimeiroNome = findViewById( R.id.editPrimeiroNome );
        editSobrenome = findViewById( R.id.editSobrenome );
        editCursoDesejado = findViewById( R.id.editCursoDesejado );
        editNumeroContato = findViewById( R.id.editNumeroContato );

        // Settando valores aos campos
        editPrimeiroNome.setText( pessoa.getNome() );
        editSobrenome.setText( pessoa.getSobrenome() );
        editCursoDesejado.setText( pessoa.getCursoDesejado() );
        editNumeroContato.setText( pessoa.getTelefoneContato() );

        // Mapeando os botões
        btnLimpar = findViewById( R.id.btnLimpar );
        btnSalvar = findViewById( R.id.btnSalvar );
        btnFinalizar = findViewById( R.id.btnFinalizar );

        btnLimpar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                editPrimeiroNome.setText("");
                editSobrenome.setText("");
                editCursoDesejado.setText("");
                editNumeroContato.setText("");
                pessoaController.limpar();

            }
        });

        btnFinalizar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Toast.makeText(MainActivity.this, "Volte sempre", Toast.LENGTH_LONG).show();
                finish();
            }
        });

        btnSalvar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                pessoa.setNome(editPrimeiroNome.getText().toString());
                pessoa.setSobrenome(editSobrenome.getText().toString());
                pessoa.setCursoDesejado(editCursoDesejado.getText().toString());
                pessoa.setTelefoneContato(editNumeroContato.getText().toString());

                Toast.makeText(MainActivity.this, "Salvo: " + pessoa.toString(), Toast.LENGTH_LONG).show();

                pessoaController.salvar( pessoa );

            }

        });

    }

}

File: GastoViagem/app/src/main/java/devandroid/thalles/gastoviagem/MainActivity.kt
package devandroid.thalles.gastoviagem

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import devandroid.thalles.gastoviagem.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity(), View.OnClickListener {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.buttonCalculate.setOnClickListener(this)

    }

    override fun onClick(view: View) {

        if( view.id == R.id.button_calculate ) {
            calculate()
        }

    }

    private fun calculate() {

        var distance = 0f
        var price = 0f
        var autonomy = 0f
        var totalValue = 0f

        if( isValid() ) {

            distance = binding.editDistance.text.toString().toFloat()
            price = binding.editPrice.text.toString().toFloat()
            autonomy = binding.editAutonomy.text.toString().toFloat()
            totalValue = (distance * price) / autonomy

            binding.textTotalValue.text = "R$ ${"%.2f".format(totalValue)}"

        } else {
            Toast.makeText( this, R.string.validation_fill_all_fields, Toast.LENGTH_SHORT ).show()
        }

    }

    private fun isValid(): Boolean {

        return binding.editDistance.text.toString() != "" &&
                binding.editPrice.text.toString() != "" &&
                binding.editAutonomy.text.toString() != "" &&
                binding.editAutonomy.text.toString().toFloat() != 0f
                binding.editDistance.text.toString().toFloat() != 0f
                binding.editPrice.text.toString().toFloat() != 0f

    }
}